src/Lecture2.hs

```diff
@@ -150,15 +171,43 @@ You're free to define any helper functions.
        treasure besides gold (if you already haven't done this).
 -}
 
+newtype Health = MkHealth Int
+newtype Attack = MkAttack Int 
+newtype Endurance = MkEndurance Int 
 -- some help in the beginning ;)
 data Knight = Knight
-    { knightHealth    :: Int
-    , knightAttack    :: Int
-    , knightEndurance :: Int
+    { knightHealth    :: Health
+    , knightAttack    :: Attack
+    , knightEndurance :: Endurance
     }
+data Color = Red
+    |Green
+    |Black
+
+
+newtype FirePower = MkFirePower Int 
+
+data Dragon = Dragon
+    { dragonType :: Color,
+      dragonFirePower :: FirePower
+    }  
```

Comment:
Your types for this task are great so far! 👏🏻 

One thing that I can't see yet is how you're going to calculate `Gold` and `Treasure` after a win. Are you planning to write a function that pattern matches on a dragon `Color` and specifies the static reward only based on color? 🤔

Reply:
As I understood it will be based only on color

Reply:
When I wrote the task I was thinking more in lines of "dragon stores the treasure inside" so the `getReward` function doesn't need to hardcode the reward based on colour. It can just extract it from the dragon.
But it's totally possible that this wasn't clear from the task 😅 
Feel free to implement your solution on getting the reward based on colour only 👍🏻